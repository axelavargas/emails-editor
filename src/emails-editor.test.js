import EmailsEditor from './index'

describe("Emails editor", () => {
    let input;

    beforeEach(() => {
        input = document.createElement('div');
        document.body.appendChild(input);
    })

    afterEach(() => {
        document.body.removeChild(input);
    })

    it("should initialize the container", () => {
        EmailsEditor(input);
        expect(input).toMatchSnapshot()
    })

    it('should allow to add and return the stored emails', () => {
        const editor = EmailsEditor(input);
        editor.add('testinvalid');
        editor.add('testinvalid2');
        editor.add('test@testme.com');
        editor.add('test2@testme.com');
        expect(editor.getEmails().length).toBe(4)
    })

    it('should mark valid emails with tag', () => {
        const editor = EmailsEditor(input);
        editor.add('test@testme.com');
        expect(editor.getEmails().filter(e => !!e.isValid).length).toBe(1);

    })
    it('should mark invalid emails', () => {
        const editor = EmailsEditor(input);
        editor.add('testinvalid');
        expect(editor.getEmails().filter(e => !!e.isValid).length).toBe(0);
    })

    it('should mix valid and invalid emails', () => {
        const editor = EmailsEditor(input);
        editor.add('testinvalid');
        editor.add('testinvalid2');
        editor.add('test@testme.com');
        editor.add('test2@testme.com');
        expect(editor.getEmails().filter(e => !!e.isValid).length).toBe(2);
        expect(editor.getEmails().filter(e => !e.isValid).length).toBe(2);
    })
})