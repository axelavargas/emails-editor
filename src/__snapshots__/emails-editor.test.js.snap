// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emails editor should initialize the container 1`] = `
<div
  class="emailsEditor"
>
  <div
    class="emailsEditor__list"
  >
    <input
      autofocus=""
      class="emailsEditor__input"
    />
  </div>
</div>
`;
